src/rfc9110.ts

```diff
@@ -7,7 +7,8 @@ export function parseWWWAuthenticate(header: string): string[] {
     const ALPHA = 'A-Za-z';
     const DIGIT = '0-9';
     const tokenChar = `!#$%&'*+\\-\\.^_\`|~${DIGIT}${ALPHA}`;
-    const tchar = `[${tokenChar}]`;
+    const nonRFCCompliantTokenCharFoundInTheWild = '=';
```

Comment:
A header with a stray ` =` at the end still does not parse correctly for me. And the parser correctly roots out `=` within a token. What is the special case you want to allow for? Perhaps a comment of what "why" above this would be helpful to others passing this way in future.

Reply:
do you have the sample header you used?

Reply:
my test token is the one in `complicated`

```
PrivateToken challenge=AAIADmlzc3Vlci5leGFtcGxlIIo-g6M9mABdLzC-9Bn6a_TNXGAF42sShbu0zNQPpLODAA5vcmlnaW4uZXhhbXBsZQ==, token-key=MIIBUjA9BgkqhkiG9w0BAQowMKANMAsGCWCGSAFlAwQCAqEaMBgGCSqGSIb3DQEBCDALBglghkgBZQMEAgKiAwIBMAOCAQ8AMIIBCgKCAQEAyxrta2qV9bHOATpM_KsluUsuZKIwNOQlCn6rQ8DfOowSmTrxKxEZCNS0cb7DHUtsmtnN2pBhKi7pA1I-beWiJNawLwnlw3TQz-Adj1KcUAp4ovZ5CPpoK1orQwyB6vGvcte155T8mKMTknaHl1fORTtSbvm_bOuZl5uEI7kPRGGiKvN6qwz1cz91l6vkTTHHMttooYHGy75gfYwOUuBlX9mZbcWE7KC-h6-814ozfRex26noKLvYHikTFxROf_ifVWGXCbCWy7nqR0zq0mTCBz_kl0DAHwDhCRBgZpg9IeX4PwhuLoI8h5zUPO9wDSo1Kpur1hLQPK0C2xNLfiJaXwIDAQAB,unknownChallengeAttribute=ignore-me, max-age=10
```

it parses with `PrivateToken.parseMultiple(header)`.

note: there is some discrepencies between `parse` and `parseMultiple`. The former should not be presented with `authScheme` (`Privatetoken ` in PP case)

Reply:
I suggest to avoid confusion to rename:
parseMultiple -> parse // make this function public.
parse -> parseSingle // make this function private.

Also, I suggest to do the right parsing here.
And to dedicate a special function to support incompatibilities. Ideally, no incompatibilities should be supported.
So, parse with the right method, and fallback to the variant version if the first parsing fails.

Reply:
Change in the last commit should address this.

* `parseMultiple` -> `parse`
* `parse` -> `private parseSingle`
* `parse` defaults to compliant tokens. If none is found, try again with non compliant tokens
* Updates `rfc9110.ts` to expose two methods `parseWWWAuthenticate` and `parseWWWAuthenticateNonCompliantToken`. The name of the later is long on purpose to be clear about the behavior. Compliance is not an option on the method because this would incentivise people to use it. Making a dedicated method helps spot it out.